test/features/ui/render/scene_test.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flyde/features/ui/render/scene.dart';
+import 'package:flyde/features/ui/render/widget.dart';
+import 'package:test/test.dart';
+
+import '../../../helpers/mocks/mock_widget.dart';
+import '../../../helpers/value_hook.dart';
+
+void main() {
+  late Scene scene;
+  late IOSink fakeOutStream;
+  late StreamController<List<int>> controller;
+  late List<Widget> widgets;
+  late State<String> widgetState;
+  const int virtualTerminalWidth = 200;
+
+  setUp(() {
+    widgetState = State('state1');
+    widgets = [
+      MockWidget(widgetState),
+      MockWidget(State('state2')),
+    ];
+    controller = StreamController();
+    fakeOutStream = IOSink(controller.sink);
+    scene = Scene(
+      widgets,
+      output: fakeOutStream,
+      fallbackTerminalColumns: virtualTerminalWidth,
+    );
+  });
+
+  test('Renders widgets', () async {
+    final hook = VHook<List<String>>([]);
+
+    controller.stream.listen((event) {
+      hook.update(
+        (previous) => [
+          ...previous,
+          String.fromCharCodes(event),
+        ],
+      );
+    });
+
+    scene.show();
+    await hook.awaitValue(
+      timeout: Duration(seconds: 1),
+      condition: (cnt) => cnt.length == 4,
+    );
+
+    hook.expect(hasLength(4));
```

Comment:
For better readability of the test. I think it's easier to see that we expect four item when it's an explicit test case. Resolve if you agree otherwise write 'No more four' so I remove it :)

Reply:
You are right, readability matters with test cases. Furthermore we dont have to consider performance at all here, so yes, keep it :)